src/acvm_interop/pwg.rs

```diff
@@ -11,185 +12,297 @@ use crate::scalar_mul::ScalarMul;
 use crate::schnorr::SchnorrSig;
 use crate::Barretenberg;
 
-use blake2::{Blake2s, Digest};
-
 mod merkle;
 
 impl PartialWitnessGenerator for Barretenberg {
-    fn solve_black_box_function_call(
+    fn aes(
+        &self,
+        _initial_witness: &mut BTreeMap<Witness, FieldElement>,
+        _inputs: &[FunctionInput],
+        _outputs: &[Witness],
+    ) -> Result<OpcodeResolution, OpcodeResolutionError> {
+        Err(OpcodeResolutionError::UnsupportedBlackBoxFunc(
+            BlackBoxFunc::AES,
+        ))
+    }
+
+    fn and(
         &self,
         initial_witness: &mut BTreeMap<Witness, FieldElement>,
-        func_call: &BlackBoxFuncCall,
+        inputs: &[FunctionInput],
+        outputs: &[Witness],
     ) -> Result<OpcodeResolution, OpcodeResolutionError> {
-        match func_call.name {
-            BlackBoxFunc::SHA256 => hash::sha256(initial_witness, func_call),
-            BlackBoxFunc::Blake2s => hash::blake2s(initial_witness, func_call),
-            BlackBoxFunc::Keccak256 => hash::keccak256(initial_witness, func_call),
-            BlackBoxFunc::EcdsaSecp256k1 => {
-                signature::ecdsa::secp256k1_prehashed(initial_witness, func_call)
-            }
-
-            BlackBoxFunc::AND | BlackBoxFunc::XOR => {
-                logic::solve_logic_opcode(initial_witness, func_call)
-            }
-            BlackBoxFunc::RANGE => range::solve_range_opcode(initial_witness, func_call),
-            BlackBoxFunc::AES => Err(OpcodeResolutionError::UnsupportedBlackBoxFunc(
-                func_call.name,
-            )),
-            BlackBoxFunc::ComputeMerkleRoot => {
-                let mut inputs_iter = func_call.inputs.iter();
-
-                let _leaf = inputs_iter.next().expect("expected a leaf");
-                let leaf = witness_to_value(initial_witness, _leaf.witness)?;
-
-                let _index = inputs_iter.next().expect("expected an index");
-                let index = witness_to_value(initial_witness, _index.witness)?;
-
-                let hash_path: Result<Vec<_>, _> = inputs_iter
-                    .map(|input| witness_to_value(initial_witness, input.witness))
-                    .collect();
-
-                let computed_merkle_root = merkle::compute_merkle_root(
-                    |left, right| self.compress_native(left, right),
-                    hash_path?,
-                    index,
-                    leaf,
+        logic::and(
```

Comment:
I think the ACVM functions need to be changed. It seems weird to be building this struct to call them.

Reply:
Yep, I made https://github.com/noir-lang/acvm/pull/268 to address this.